pyopensprinkler/program.py

```diff
@@ -67,6 +68,9 @@ def _get_data_flag_bits(self):
     def _set_data_flag_bit(self, index, value):
         print("foo")
 
+    def add_on_removed_observer(self, observer):
```

Comment:
technically it is both add and remove so the name is a bit off

Reply:
This is actually just remove. Chicken/egg for added.

Reply:
Meaning if you’re adding a callback for added on a program...the program is already added (in the context of a program instance).

Reply:
yea but you have both "added" and "removed" "events". Maybe just `add_observer`?

Reply:
The event on the program directly is only removed..